src/coreclr/vm/comutilnative.cpp

```diff
@@ -801,26 +801,31 @@ extern "C" int QCALLTYPE GCInterface_EndNoGCRegion()
 **Arguments: args->handle -- the OBJECTHANDLE to the object which we're locating.
 **Exceptions: ArgumentException if handle points to an object which is not accessible.
 ==============================================================================*/
-FCIMPL1(int, GCInterface::GetGenerationWR, LPVOID handle)
+extern "C" int QCALLTYPE GCInterface_GetGenerationWR(LPVOID handle)
```

Comment:
This QCall can be deleted and moved to C#. Something like:

```
        public static int GetGeneration(WeakReference wo)
        {
            // Note - This throws an NRE if given a null weak reference.
            object? obj = GCHandle.InternalGet(wo.WeakHandle);
            KeepAlive(wo);

            if (obj == null)
            {
                throw new ArgumentNullException(nameof(wo));
            }

            return GetGeneration(obj);
        }
```

Reply:
So I was going to that actually, but I didn't want to extend the lifetime of the object by extracting it from the `WeakReference`. Does that not matter much in this case?

Reply:
I do not think it matters in this case.

This API is useless, it has 0 known uses https://apisof.net/catalog/364d5dc6-3461-d876-bdc8-8a51a38b6606.